bin/utils/bulk_compression.py

```diff
@@ -0,0 +1,418 @@
+
+    print("║")
+    print("║ You may choose to OVERWRITE or SKIP all of these.")
+    if (crash_protection_type == 2):
+        print("║ NOTE: chdman cannot overwrite .iso files. These will be skipped regardless.")
+    choice = input("║ Press 'O' to overwrite or 'S' to skip and press ENTER: ").lower()
+
+    if (choice in dupe_options):
+        if (not dupe_options[choice]):                              # Skip
+            [dupe_names.append(fname[:-4]) for fname in dupe_files if fname[:-4] not in dupe_names]
+        return dupe_names
+    else:
+        exitInvalidOption()
+
+# =================================================================================================
+
+def printInitialStatus(decompressing, target_fname):
+
+    if (gamecount[0] != 0):
+        print("╟-------------------------------------------------------------------------------╢")
+    gamecount[0] += 1
+
+    if (decompressing):
+        print(f"║ Extracting to {target_fname}... ({gamecount[0]}){(58 - len(target_fname) - len(str(gamecount[0]))) * ' '}║")
+    else:
+        print(f"║ Compressing to {target_fname}... ({gamecount[0]}){(57 - len(target_fname) - len(str(gamecount[0]))) * ' '}║")
+
+# -------------------------------------------------------------------------------------------------
+
+def printSkip(target_fname):
+
+    if (gamecount[0] != 0):
+        print("╟-------------------------------------------------------------------------------╢")
+    gamecount[0] += 1
+    print(f"║ Skipping creation of {target_fname}{(57 - len(target_fname)) * ' '}║")
+
+# =================================================================================================
+
+def createCommandList(mode, source_fname, target_fname, blocksize=0):
+
+    match mode:
+        case 1:
+            return [["maxcso", f"--block={blocksize}", source_fname]]
+        case 2:
+            return [["chdman", "createraw", "-us", "2048", "-hs", f"{blocksize}", "-f", "-i", source_fname, "-o", target_fname]]
+        case 3:
+            return [["chdman", "createcd", "-hs", f"{blocksize}", "-i", source_fname, "-o", f"{source_fname[:-4]}.chd"]]
+        case 4:
+            return [["maxcso", "--decompress", source_fname],
+                    ["chdman", "createraw", "-us", "2048", "-hs", f"{blocksize}", "-f", "-i", f"{source_fname[:-4]}.iso", "-o", f"{source_fname[:-4]}.chd"]]
+        case 5:
+            return [["chdman", "extractraw", "-i", source_fname, "-o", f"{source_fname[:-4]}.iso"],
+                    ["maxcso", f"--block={blocksize}", f"{source_fname[:-4]}.iso"]]
+        case 6:
+            return [["chdman", "extractraw", "-i", source_fname, "-o", target_fname]]
+        case 7:
+            return [["chdman", "extractcd", "-i", source_fname, "-o", target_fname]]
+        case 8:
+            return [["maxcso", "--decompress", source_fname]]
+        case _:
+            print("You definitely should not be here right now.")
+            sys.exit(1)
+
```

Comment:
I’ve been using `createdvd` with my games. Just wondering why `createraw` is used here - is there a benefit, or much of a much ess?

Reply:
I made the script before createdvd was an option, but also I find createraw tends to give slightly better compression, from my own tests.

Reply:
Also they share similar options, createraw has more versatility in the way it has more options like createdvd doesn't have  `No --unitsize Option:'`

Both have the same compression types (default is lzma,zlib,huff,flac).

But with createdvd you can easier give the correct hunksize. Createraw will default to correct one but you can change to whatever for more or less compression.

Generally looking at the options they do have the same compression powers just differently handled with the code logic.